### DISCUSSION TITLE
Configure username and password for kafka-bridge external endpoint

### DISCUSSION THREAD

**Discussion by rushikeshkulat** [strimzi/strimzi-kafka-operator#7565] [Q&A]
Hi , 
I want to configure username and password for kafka-bridge external endpoint. is it possible at bridge level or do i need to use reverse proxy for this .

i have deplyed kafka bridge using bewlow template.

apiVersion: kafka.strimzi.io/v1beta2
kind: KafkaBridge
metadata:
  name: my-bridge-test
spec:
  replicas: 1
  bootstrapServers: a541deb5e6c9a497bbc41a7666967f63-642738248.eu-north-1.elb.amazonaws.com:31002
  tls:
    trustedCertificates:
    - secretName: kafka-cluster-cluster-ca-cert
      certificate: ca.crt
  authentication:
    type: tls
    certificateAndKey:
      secretName: adminuser
      certificate: user.crt
      key: user.key
  http:
    port: 8080

**Comment by scholzj** [comment]
What do you mean with _external endpoint_? The HTTP endpoint? To secure that, you should front the bridge with some API Gateway or proxy. The Bridge itself does to provide any security on it.

**Reply by rushikeshkulat** [reply]
yes HTTP end point . thanks for feedback

**Reply by rushikeshkulat** [reply]
I have fronted my kafkabridge with cloudflare proxy. 

when I'm trying to list topics via web browser, I'm able to list.

https://********************.com/topics/

["json_topic","__strimzi_store_topic","kafka-connect-status","GSMDPKafkaTopic","kafka-connect-offsets","test-topic","bridge-quickstart-topic","__strimzi-topic-operator-kstreams-topic-store-changelog","my-first-topic","kafka-connect-configs"]

__________________________________

but when i try to list using CURL command I'm getting 400 Bad Request

curl -X -v GET https://**********************.com/topics

curl: (6) Could not resolve host: GET
<html>
<head><title>400 Bad Request</title></head>
<body>
<center><h1>400 Bad Request</h1></center>
<hr><center>cloudflare</center>
</body>
</html>

**Reply by ppatierno** [reply]
Are you sure the request has reached the bridge? Is that a response from cloudflare proxy? You could double check on the bridge log as well.

**Reply by rushikeshkulat** [reply]
when I hit URL from browser I can see logs for that request, status 200 ok.
But when I try using curl, I don't see any bridge log

**Reply by ppatierno** [reply]
So it means that your proxy is not forwarding the request to the bridge when usiong cURL. You should understand why. There is nothing bad with the bridge in this case.

**Reply by rushikeshkulat** [reply]
thanks for help
